### PR TITLE
[9.0][ADD][sale_order_cancel_invoice] Cancel draft invoices when cancelling sale order

### PR DESCRIPTION
**Pull request by BenjaHe**
This module cancel all invoices generated by a Sale order when we cancel this sale order AND those invoices have no number AND are in state draft, proforma or proforma2

### CODE REVIEW

File: sale_order_cancel_invoice/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# -*- coding: utf-8 -*-
+from . import models
```
Comment (JayVora-SerpentCS) [inline sale_order_cancel_invoice/__init__.py]: Add # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).

File: sale_order_cancel_invoice/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# -*- coding: utf-8 -*-
+from . import sale_order
```
Comment (JayVora-SerpentCS) [inline sale_order_cancel_invoice/models/__init__.py]: same as above

File: sale_order_cancel_invoice/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    @api.multi
+    def action_cancel(self):
+        for order in self:
+            order.cancel_draft_invoice()
+        return super(SaleOrder, self).action_cancel()
+
+    def cancel_draft_invoice(self):
+        for invoice in self.invoice_ids:
+            if not invoice.number and (invoice.state == 'draft' or invoice.state == 'proforma' or invoice.state == 'proforma2'):
+                invoice.signal_workflow('invoice_cancel')
```
Comment (JayVora-SerpentCS) [inline sale_order_cancel_invoice/models/sale_order.py]: check pylint plz.

File: sale_order_cancel_invoice/models/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    @api.multi
+    def write(self, values):
+        for order in self:
+            if 'order_line' in values:
+                order.cancel_draft_invoice()
+        res = super(SaleOrder, self).write(values)
+        return res
```
Comment (JayVora-SerpentCS) [inline sale_order_cancel_invoice/models/sale_order.py]: return super(SaleOrder, self).write(values) will do

**Comment by JayVora-SerpentCS** [comment]
Fix travis please @BenjaHe

**Comment by rafaelbn** [comment]
Hi @BenjaHe are you going to fix this PR please? Thanks!

**Comment by rousseldenis** [comment]
Closing as very old. Feel free to reopen it if needed.

**Pull request closed without merging**